Assignment 5.Rmd

```diff
@@ -40,42 +38,54 @@ We want to see if we can build a decision tree to help teachers decide which stu
 
 #Visualize our outcome variable "score"
 ```{r}
-
+hist(D1$score)
 ```
 
 #Create a categorical outcome variable based on student score to advise the teacher using an "ifelse" statement
 ```{r}
-D1$advice <- 
+D1$advice <- ifelse(D1$score <= .4, "teacher should intervene", ifelse(D1$score > .8, "no action", "teacher should monitor student progress"))
 ```
 
 #Build a decision tree that predicts "advice" based on how many problems students have answered before, the percentage of those problems they got correct and how many hints they required
 ```{r}
-score_ctree <- 
+D1$advice <- factor(D1$advice)
+score_ctree <- ctree(advice ~ prior_prob_count + prior_percent_correct + hints, data = D1)
+
 ```
 
 #Plot tree
 ```{r}
-
+plot(score_ctree)
 ```
 
 Please interpret the tree, which two behaviors do you think the teacher should most closely pay attemtion to?
 
+The teacher should most closely pay attention to students whose prior_percent_correct is less than or equal to .629 and those who ask for more than 12 hints.
+
 #Test Tree
 Upload the data "intelligent_tutor_new.csv". This is a data set of a differnt sample of students doing the same problems in the same system. We can use the tree we built for the previous data set to try to predict the "advice" we should give the teacher about these new students. 
 
 ```{r}
 #Upload new data
 
-D2 <- 
+D2 <- read.csv("intelligent_tutor_new.csv", header = TRUE)
 
 #Generate predicted advice using the predict() command for new students based on tree generated from old students
 
-D2$prediction <- 
+D2$prediction <- predict(score_ctree, D2)
 
 ``` 
 ## Part III
 Compare the predicted advice with the actual advice that these students recieved. What is the difference between the observed and predicted results?
 
+```{r}
+D2$advice <- ifelse(D2$score <= .4, "teacher should intervene", ifelse(D2$score > .8, "no action", "teacher should monitor student progress"))
+D2$mismatch <- ifelse(D2$prediction != D2$advice, 1, 0)
+mean_difference <- mean(D2$mismatch)
+
+```
+
```

Comment:
Overall, great job on the coding part of this assignment but you may want to work on the analysis. As in, specify whether the model overfits or whether it is generalizable in terms of its accuracy rate.